bin/dfx-software-nns-dapp-version.test

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+
+(
+  echo "Using --pinned should get a release from the versions file"
+  ACTUAL_RELEASE="$(dfx-software-nns-dapp-version --pinned)"
+  EXPECTED_RELEASE="$(. "$SOURCE_DIR/versions.bash" && echo "${NNS_DAPP_RELEASE}")"
+  [[ "$ACTUAL_RELEASE" =~ ^proposal-[0-9]*$ ]] || {
+    echo "ERROR: The release is not a proposal tag: '$ACTUAL_RELEASE'"
```

Comment:
I think it's good practice for such messages to say "... should ...".
Otherwise it can be unclear whether the message describes the correct or the incorrect state.
So:
ERROR: The release should be a proposal tag: '$ACTUAL_RELEASE'